opamp/observiq/observiq_packages_state_provider.go

```diff
@@ -0,0 +1,174 @@
+func (p *PackagesStateProvider) Packages() ([]string, error) {
+	p.logger.Debug("Retrieve package names")
+
+	return nil, errors.New("method not implemented: PackageStateProvider Packages")
+}
+
+// PackageState not implemented so returns an error with this info
+func (p *PackagesStateProvider) PackageState(_ string) (state types.PackageState, err error) {
+	p.logger.Debug("Retrieve package state")
+
+	packageState := types.PackageState{}
+
+	return packageState, errors.New("method not implemented: PackageStateProvider PackageState")
+}
+
+// SetPackageState not implemented so returns an error with this info
+func (p *PackagesStateProvider) SetPackageState(_ string, _ types.PackageState) error {
+	p.logger.Debug("Set package state")
+
+	return errors.New("method not implemented: PackageStateProvider SetPackageState")
+}
+
+// CreatePackage not implemented so returns an error with this info
+func (p *PackagesStateProvider) CreatePackage(_ string, _ protobufs.PackageAvailable_PackageType) error {
+	p.logger.Debug("Create package")
+
+	return errors.New("method not implemented: PackageStateProvider CreatePackage")
+}
+
+// FileContentHash not implemented so returns an error with this info
+func (p *PackagesStateProvider) FileContentHash(_ string) ([]byte, error) {
+	p.logger.Debug("Retrieve package content hash")
+
+	return nil, errors.New("method not implemented: PackageStateProvider FileContentHash")
+}
+
+// UpdateContent not implemented so returns an error with this info
+func (p *PackagesStateProvider) UpdateContent(_ context.Context, _ string, _ io.Reader, _ []byte) error {
+	p.logger.Debug("Update package content")
+
+	return errors.New("method not implemented: PackageStateProvider UpdateContent")
+}
+
+// DeletePackage not implemented so returns an error with this info
+func (p *PackagesStateProvider) DeletePackage(_ string) error {
+	p.logger.Debug("Delete package")
+
+	return errors.New("method not implemented: PackageStateProvider DeletePackage")
+}
+
+// LastReportedStatuses retrieves the PackagesStatuses from a saved yaml file
+func (p *PackagesStateProvider) LastReportedStatuses() (*protobufs.PackageStatuses, error) {
+	p.logger.Debug("Retrieve last reported package statuses")
+
+	packages := map[string]*protobufs.PackageStatus{
+		"observiq-otel-collector": {
+			Name:            "observiq-otel-collector",
+			AgentHasVersion: version.Version(),
+			Status:          protobufs.PackageStatus_Installed,
+		},
+	}
+	packageStatuses := &protobufs.PackageStatuses{
+		Packages: packages,
+	}
+
+	// If there's a problem reading the file, we just return a barebones PackageStatuses.
+	if _, err := os.Stat(p.yamlPath); errors.Is(err, os.ErrNotExist) {
+		p.logger.Debug("Package statuses yaml doesn't exist")
+		return packageStatuses, nil
+	}
+
+	statusesBytes, err := os.ReadFile(p.yamlPath)
+	if err != nil {
+		return packageStatuses, fmt.Errorf("problem reading from package statuses yaml: %w", err)
```

Comment:
nit: every other error starts with `failed to X` might be worth keeping this error consistent.